project1_tiny_calculator/lib/calculate_provider.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+
+class CalculateProvider extends ChangeNotifier {
+  List<String> operations = ['+', '/', '-', '*'];
+  final TextEditingController firstNumberController = TextEditingController();
+  final TextEditingController secondNumberController = TextEditingController();
+  String result = '';
+  int currentIndex = 0;
+
+  String get currentOperation => operations[currentIndex];
+
+  void changeOperator(isUp) {
+    currentIndex = (currentIndex + (isUp ? 1 : -1)) % operations.length;
+    if (currentIndex < 0) currentIndex += operations.length;
+
+    calculate();
+    notifyListeners();
```

Comment:
this looks redundant, we don't need it

Reply:
done

Reply:
notifyListner() is been removed from here .